### PR TITLE
144 api for granting permission

### PR DESCRIPTION
**Pull request by anushkavidanage**
# Pull Request Details

## What issue does this PR address

- UI for granting permissions

- Link to associated issue:  https://github.com/anusii/solidpod/issues/144

## Checklist

Complete the check-list below to ensure your branch is ready for PR.

Flutter Style Guide: https://survivor.togaware.com/gnulinux/flutter-style.html

- [x] Screenshots included in linked issue
- [x] Changes adhere to the team style and coding guideline
- [x] No confidential information
- [x] No duplicated content
- [x] No lint check errors related to your changes (`make prep` or `flutter analyze lib`)
- [x] Pre-exisiting lint errors noted: [HERE]
- [x] Tested on at least one device
  - [ ] Android Phone
  - [ ] Android Emulator
  - [ ] Chrome on Android
  - [x] Chrome
  - [ ] iOS
  - [x] Linux
  - [ ] MacOS
  - [x] Windows
- [x] Added 2 reviewers (or 1 for private repositories then they add another)

## Finalising

Once PR discussion is complete and 2 reviewers have approved:

- [ ] Merge dev into the branch
- [ ] Resolve any conflicts
- [ ] Add one line summary into CHANGELOG.md
- [ ] Bump appropriate version number in pubspec.yaml
- [ ] Push to git repository and review
- [ ] Merge PR into dev

### CODE REVIEW

File: lib/screens/sharing.dart [hunk 1]
```diff
@@ -0,0 +1,518 @@
...
+  // Form controllers
+  final formKey = GlobalKey<FormState>();
+  final formControllerWebId = TextEditingController();
+
+  @override
+  void initState() {
+    super.initState();
+  }
+
+  Future<void> _showPrivateData(String title) async {
```
Comment (cdawei) [inline lib/screens/sharing.dart]: This function is not used, shall we remove it?

File: lib/screens/sharing.dart [hunk 2]
```diff
@@ -0,0 +1,518 @@
...
+        setState(() {
+          // End loading.
+
+          _isLoading = false;
+        });
+      }
+    }
+  }
+
+  Future<void> _writePrivateData() async {
```
Comment (cdawei) [inline lib/screens/sharing.dart:81]: This function is not used, shall we remove it?

**Review by cdawei — APPROVED** [review]
Worked as advertised, noted the comments here https://github.com/anusii/solidpod/issues/144

**Review by junhaow1 — APPROVED** [review]
works in my test

**Pull request merged**